ui/lib/core/addon/components/secret-list-header.hbs

```diff
@@ -25,6 +25,80 @@
       <h1 class="title is-3">
         <Icon @name={{@model.icon}} @size="24" class="has-text-grey-light" />
         {{@model.id}}
+
+        {{#if this.isPki}}
+          {{#if @isEngine}}
+            <LinkToExternal @route="list" class="tag is-borderless">
+              <Icon @name="arrow-left" />
+              Return to old PKI
+            </LinkToExternal>
+          {{else}}
+            {{#if this.shouldHideBetaModal}}
+              <LinkTo
+                @route="vault.cluster.secrets.backend.pki.overview"
+                class="has-background-success-light tag is-borderless"
+              >
+                <Icon @name="key" />
+                New PKI UI available
+              </LinkTo>
+            {{else}}
+              <button
+                type="button"
+                class="is-borderless has-background-success-light"
+                {{on "click" (action (mut this.modalOpen) true)}}
+              >
+                <Icon @name="key" />
+                New PKI UI available
+              </button>
+              <Modal
+                @title="PKI Beta"
+                @onClose={{action (mut this.modalOpen) false}}
+                @isActive={{this.modalOpen}}
+                @showCloseButton={{true}}
+              >
+                <section class="modal-card-body">
+                  Information about the PKI beta and what it contains
+                  <ul>
+                    <li>Feature</li>
+                    <li>Feature</li>
+                    <li>Feature</li>
+                  </ul>
+                  <br />
+                  Remind the user that they can always go back and forth between the two.
+                  <br />
+                  <br />
+                  You can also copy the following URL and bookmark it to always go to the beta version:
+                  <code>http://localhost:8200/ui/vault/secrets/{{@model.id}}/pki/overview</code>
+                  <br />
+                  <br />
+                  <input
+                    type="checkbox"
+                    name="hide beta modal"
+                    checked={{this.key.enterAsText}}
+                    onchange={{this.toggleHideBetaModal}}
+                  />
+                  <label for="hide-beta-modal">
+                    Don't show me this again
+                  </label>
+                </section>
+                <footer class="modal-card-foot modal-card-foot-outlined">
+                  <button type="button" class="button is-primary" {{on "click" this.transitionToNewPki}}>
+                    Go to beta
+                  </button>
+                  <button
+                    type="button"
+                    class="button is-secondary"
+                    {{on "click" (action (mut this.modalOpen) false)}}
+                    data-test-metrics-config-cancel
+                  >
+                    Cancel
+                  </button>
+                </footer>
+              </Modal>
```

Comment:
I wanted to move all this to a component, since it's a modal for pki beta and doesn't necessarily have anything to do with other engines. I wasn't sure where it would live since there aren't specific engine folders? We have the pki engine page components folder, but it also doesn't feel right to put this there too.

Reply:
I was thinking about this too. I don't really love how much logic was added to allow it to be used in engines in the first place and this would compound that. I think it's fine since this will likely only live for a release cycle and then be removed.

To offer an alternate option, you could move this to a component in the core addon named something like `secrets-beta` and then use args or yield for secret specific copy. Tough to say if it's worth it or if we will ever use this pattern again to beta test other secrets, but if nothing else it would be easier to remove from the `SecretListHeader` after the pki beta is finished.

Reply:
Thanks for your feedback! :) Haha I was so torn! If we're most likely only using it for this release cycle, I feel like a component isn't necessary especially if we are going to remove this eventually and future beta releases _might_ look different

Reply:
Yes I agree with that and a good chance that the component wouldn't be reusable anyhow.

Reply:
Another option is adding a {{yield}} here and then making a component in the pki engine specifically. That _would_ mean yielding it in each <SecretListHeader> which makes removing a little trickier, but moves this logic to the component. This way will probably be just fine, though - especially as it's just this release.

Reply:
I'm not sure you can use a component from an engine in an addon can you? I thought it was only the other way around.

Reply:
Oh I meant like If <SecretListHeader> accepted a block (so adding {{yield}} here) then in the PKI engine we could pass `<Pki::BetaTag>` or whatever the component would be, in each of the pki engine's SecretListHeaders